src/components/Destinations/DestinationConfigInput.tsx

```diff
@@ -0,0 +1,283 @@
+    // Check if any child specs has type object
+    const filteredChildSpecs: any = [];
+    selectedSpec?.specs.forEach((ele: any) => {
+      if (ele.parent === dropDownVal) {
+        // Check if the child has another level or not
+        if (ele?.specs && ele?.enum.length === 0) {
+          ele.specs.forEach((childEle: any) => {
+            filteredChildSpecs.push({ ...childEle, order: ele.order });
+          });
+        } else {
+          filteredChildSpecs.push(ele);
+        }
+      }
+    });
+
+    // Set the order of child specs to be displayed at correct position
+    filteredChildSpecs.forEach((ele: any) => {
+      ele.order = selectedSpec.order;
+    });
+    // Update the specs state
+
+    // Find the specs that will have parent in the following enum array
+    const enumsToRemove: string[] = selectedSpec?.enum.filter(
+      (ele: any) => ele !== dropDownVal
+    );
+
+    const tempSpecs = connectorSpecs
+      .filter((sp: any) => !enumsToRemove.includes(sp?.parent))
+      .concat(filteredChildSpecs);
+
+    // Unregister the form fields that have parent in enumsToRemove
+    connectorSpecs.forEach((sp: any) => {
+      if (enumsToRemove.includes(sp?.parent)) unregisterFormField(sp?.field);
+    });
+
+    setConnectorSpecs(tempSpecs);
+  };
+
+  useEffect(() => {
+    setConnectorSpecs(specs);
+  }, [specs]);
+
+  useEffect(() => {
+    const tempShowPasswords: any = {};
+    specs.forEach((element) => {
+      if (element?.airbyte_secret) {
+        tempShowPasswords[element.field] = false;
+      }
+    });
+    setShowPasswords(tempShowPasswords);
+  }, [specs]);
+  return (
+    <>
+      {connectorSpecs
+        .sort((input1: any, input2: any) => {
+          if (!input2.order) return -1;
+          if (!input1.order) return -1;
+          return input1.order - input2.order;
+        })
+        ?.map((spec: any, idx: number) =>
+          spec.type === 'string' ? (
+            spec?.const ? ( // type == string and a const selected value
+              <React.Fragment key={idx}>
+                <TextField
+                  sx={{ width: '100%' }}
+                  label={spec.const}
+                  variant="outlined"
+                  value={spec?.const}
+                  {...registerFormFieldValue(`${spec.field}`, {
+                    required: spec.required,
+                  })}
+                ></TextField>
+                <Box sx={{ m: 2 }} />
+              </React.Fragment>
+            ) : spec.airbyte_secret ? ( // type == string and a password/secret field
+              <React.Fragment key={idx}>
+                <TextField
+                  sx={{ width: '100%' }}
+                  label={spec?.title}
+                  variant="outlined"
+                  type={showPasswords[`${spec.field}`] ? 'text' : 'password'}
+                  {...registerFormFieldValue(`${spec.field}`, {
+                    required: spec.required,
+                  })}
```

Comment:
Check if text field can be extracted out as a separate component with spec prop and used in all the below areas.

Reply:
Will do it as part of the cleanup issue. Can you add this there ?